src/components/structures/TimelinePanel.js

```diff
@@ -706,7 +735,7 @@ var TimelinePanel = React.createClass({
 
         // the messagePanel doesn't know where the read marker is.
         // if we know the timestamp of the read marker, make a guess based on that.
-        var rmTs = TimelinePanel.roomReadMarkerTsMap[this.props.timelineSet.roomId];
+        const rmTs = TimelinePanel.roomReadMarkerTsMap[this.props.timelineSet.room.roomId];
```

Comment:
Looks like it! This is reached if `MessagePanel` has _not_ got the related event paginated in.

It doesn't look like much is done with the result of `TimelinePanel.getReadMarkerPosition()` other than checking that it's in the view-port when `updateReadMarker` is called (which might explain why we were seeing it following when scrolling down) and specifying whether it's visible to `MessagePanel`.